### ISSUE TITLE
pass job output to another job

### ISSUE THREAD

**Issue by MoatazAbdAlmageed** [expo/expo-github-action#176]
### Topic and scope of discussion
pass job output to another job

### Motivation
after pass those output I will use it to send it via email

### Additional context

I want to get job output like `{{ steps.releaseChannel.outputs.value }}`
I created id with name `android` and in email job I tries to access it and not working  you can found it in https://github.com/MoatazAbdAlmageed/First-Mobile-App/blob/main/.github/workflows/main.yml 

so I want to pass the following result to  `send-bundle-via-email` job

```

Run expo build:android
- Making sure project is set up correctly...

[13:11:16] expo build:android has been superseded by eas build. Learn more: https://blog.expo.dev/turtle-goes-out-to-sea-d334db2a6b60

[13:11:16] Run the following:

[13:11:16] › npm install -g eas-cli
[13:11:16] › eas build -p android https://docs.expo.dev/build/setup/

[13:11:16] expo build:android will be discontinued on January 4, 2023 (274 days left).

[13:11:18] Checking if there is a build in progress...

[13:11:18] Accessing credentials for moataz.mohammady in project arab-hardware-app

[13:11:18] › Expo SDK: 44.0.0
[13:11:18] › Release channel: default
[13:11:18] › Workflow: Managed

[13:11:18] Building optimized bundles and generating sourcemaps...
[13:11:21] Starting Metro Bundler
[13:12:1[9](https://github.com/MoatazAbdAlmageed/First-Mobile-App/runs/5834070018?check_suite_focus=true#step:7:9)] Android Bundling complete 58037ms
[13:12:20] 
[13:12:20] Bundle                     Size
┌ index.ios.js           858 kB
├ index.android.js       859 kB
├ index.ios.js.map      3.36 MB
└ index.android.js.map  3.36 MB
[13:12:20] 
[13:12:20] 💡 JavaScript bundle sizes affect startup time. Learn more: https://expo.fyi/javascript-bundle-sizes
[13:12:20] 
[13:12:20] Analyzing assets
[13:12:20] Saving assets
[13:12:20] No assets changed, skipped.
[13:12:20] 
Processing asset bundle patterns:
[13:12:20] - /Users/runner/work/First-Mobile-App/First-Mobile-App/**/*
[13:12:20] 
[13:12:20] Uploading JavaScript bundles
[13:12:21] Publish complete

[13:12:21] 📝  Manifest: https://exp.host/@moataz.mohammady/arab-hardware-app/index.exp?sdkVersion=44.0.0 Learn more: https://expo.fyi/manifest-url
[13:12:21] ⚙️   Project page: https://expo.dev/@moataz.mohammady/arab-hardware-app?serviceType=classic&distribution=expo-go Learn more: https://expo.fyi/project-page

[13:12:21] Checking if this build already exists...

[13:12:22] Build started, it may take a few minutes to complete.
- 
[13:12:22] You can check the queue length at https://expo.dev/turtle-status

[13:12:22] You can monitor the build at

 https://expo.dev/accounts/moataz.mohammady/projects/arab-hardware-app/builds/a3e97e63-0a49-4cf8-8b59-68659a460f84

[13:12:22] Waiting for build to complete.
You can press Ctrl+C to exit. It won't cancel the build, you'll be able to monitor it at the printed URL.
✔ Build finished.

[13:27:32] Successfully built standalone app: https://expo.dev/artifacts/54d6843a-473f-4a01-8a0c-3[11](https://github.com/MoatazAbdAlmageed/First-Mobile-App/runs/5834070018?check_suite_focus=true#step:7:11)bc262ba[24](https://github.com/MoatazAbdAlmageed/First-Mobile-App/runs/5834070018?check_suite_focus=true#step:7:24)
```

**Comment by MoatazAbdAlmageed** [comment]
@byCedric pleas help I want to sent apk to an email

**Comment by byCedric** [comment]
Hi @MoatazAbdAlmageed! Two things are going on here:

1. If you want to send the build logs, you'll need to write the output to a file. You can do so by running expo like: `$ expo build:android > build-android.log`. This does obfuscate the output from the workflow itself, so I would recommend add another step outputting the logs from file, just to have it in the action as well (e.g. `$ cat build-android.log`). Since this is a file written to the file system, you can just add this file as attachment to your email in the other job.

2. If you want to reuse the actual outputs from github actions (which are not written to the file system), you'll need to add a few more options. Right now, you only have the `needs: [Build-xxx]`, which let github actions know it should wait on that job. But you did not specify the outputs coming from that build job. You can find a better explanation here: https://stackoverflow.com/a/61236803.

Hope this helps!

**Comment by MoatazAbdAlmageed** [comment]
Thank you @byCedric  this was helpful
